plugins/pulp_rpm/plugins/distributors/yum/metadata/updateinfo.py

```diff
@@ -90,24 +95,25 @@ def _get_package_checksum_tuple(self, package):
         :param package: package from the erratum package list
         :type  package: dict
         :return: checksum type and value to publish. An empty tuple is returned if there is
-                 no checksum of the repository checksum type available.
+                 no checksum of the requested or distributor checksum type available.
         :rtype: tuple
         """
         package_checksum_tuple = ()
-        repo_checksum_type = self.checksum_type
-        package_checksums = package.get('sum')
-        if package_checksums:
-            checksum_types = package_checksums[::2]
-            checksum_values = package_checksums[1::2]
-            checksums = dict(zip(checksum_types, checksum_values))
-            if repo_checksum_type in checksums:
-                checksum_value = checksums[repo_checksum_type]
-                package_checksum_tuple = (repo_checksum_type, checksum_value)
-        else:
-            checksum_type = package.get('type')
-            checksum_value = package.get('sums')
-            if checksum_type == repo_checksum_type and checksum_value:
+        dist_checksum_type = self.checksum_type
+        package_checksums = package.get('sum', [])
```

Comment:
A quick comment to explain that this is a list of alternating checksum types and values would help future readers understand the code below.